forc-pkg/src/pkg.rs

```diff
@@ -1720,19 +1779,45 @@ pub fn sway_build_config(
 /// then the std prelude will also be added.
 pub fn dependency_namespace(
     namespace_map: &HashMap<NodeIx, namespace::Module>,
+    compiled_map: &HashMap<NodeIx, Compiled>,
```

Comment:
Let's clarify this should only contain compiled contract deps:
```suggestion
    compiled_contract_deps: &HashMap<NodeIx, Compiled>,
```